### ISSUE TITLE
prefetch access through an authenticated proxy server  

### ISSUE THREAD

**Issue by vipints** [ncbi/sra-tools#589]
I didn't see a thread/doc describing how to use an authenticated proxy server to use sra-tools. If there are something available and if I missed it, please point me the same. 
I am using `sratoolkit.2.11.3-centos_linux64` on my HPC server where the access to outside world is restricted with outbound connections are bypassed via a proxy server and this access to the proxy server is authenticated. 

In my user setting `~/.ncbi/user-settings.mkfg`, I have provided:
```
/http/proxy/path = "http://<uname>:<code>@proxy-server.com:3128"
```
Started to use the prefetch, getting the following `stdout`:
```
[uname@login-03 ]$ /sharedutils/sratoolkit.2.11.3-centos_linux64/bin/./prefetch -v -O --force ALL SRR7236860
2022-02-11T07:47:56 prefetch.2.11.2: Current preference is set to retrieve SRA Normalized Format files with full base quality scores.
2022-02-11T07:48:56 prefetch.2.11.2: Current preference is set to retrieve SRA Normalized Format files with full base quality scores.
[uname@login-03 ]$
```
But this didn't do anything. I have requested the IT support team to whitelist the access to service `www.ncbi.nlm.nih.gov:443` 
Do I miss something here? 

thanks in advance! 
vipin

**Comment by stineaj** [comment]
There is a short description of setting the proxy information using the interactive view on this page.
https://github.com/ncbi/sra-tools/wiki/05.-Toolkit-Configuration

The options you are using in your prefetch command includes an attempt to specify the output directory (-O) but does not provide the output directory you want prefetch to use.

You can add a second -v to see additional information for the prefetch connection and to see if prefetch is connecting or failing on the proxy connection.

**Comment by klymenko** [comment]
Today we don't have support of authenticated proxy.

**Comment by tudor-turcu** [comment]
Any news on this issue? 
In some large pharma or biotech companies by default the normal users don't have direct Internet access, and they must configure a HTTP proxy, with a user/password for this..

**Comment by durbrow** [comment]
Users with such problems should ask their corporate IT for help. The problem is on their end. Sent from my iPhoneOn Mar 20, 2023, at 06:16, Tudor-Aurel Turcu ***@***.***> wrote:﻿
Any news on this issue?
In some large pharma or biotech companies by default the normal users don't have direct Internet access, and they must configure a HTTP proxy, with a user/password for this..

**Comment by durbrow** [comment]
Corp IT controls the proxy. It's their job. It's not my job. It's going to take less time and effort to get them to do it.

**Comment by tudor-turcu** [comment]
It's sad to see such poor support from a software published by US government and used by hundred of companies and research institutes, but, that's life.. :(

**Comment by ernstki** [comment]
> It's sad to see such poor support from a software published by US government and used by hundred of companies and research institutes, but, that's life.. :(

That's one take. Another possible interpretation is it's an invitation for someone from the community of users to provide a compelling initial implementation, and assist the SRA team in getting that functionality integrated.

**Comment by durbrow** [comment]
Indeed, we would appreciate the assist. It's not that we think it's a bad feature and don't want to implement it; we have not had the manpower to do it in 3 years, and there is no foreseeable change.

**Comment by ernstki** [comment]
> what can the IT do if a software does not have proper proxy support? They aren't going to change their IT policy for just one application out of hundreds, or add exceptions just for those users that happen to use SRA Toolkit, to access the proxy without a username or password.

@vipints I just became aware of [proxychains](https://github.com/haad/proxychains) today ([via](https://github.com/simplenote-vim/simplenote.vim#special-issue-concerning-gfw)).

Offhand, I cannot surmise if this would work with the SRA utilities, or if it would work for _you_ in your specific context, but certainly it's intended for this problem space.
